Install CCF on AzureVM

ross-p-smith opened the pull request:
Simple guide to install a DC Series VM in Azure and have CCF installed at boot time through a cloud-init file.

I've also added a quick description in the "build_apps" section of the docs

Resolves #4493

jumaffre commented on README.md:
```diff
@@ -14,7 +14,7 @@ and performant applications that focus on multi-party compute and data.
 - Read the [CCF overview](https://ccf.microsoft.com/) and get familiar with [CCF's core concepts](https://microsoft.github.io/CCF/main/overview/what_is_ccf.html)
 - [Install](https://microsoft.github.io/CCF/main/build_apps/install_bin.html) CCF on Linux
-- Quickly build and run the [sample CCF application](https://github.com/microsoft/ccf-app-template)
+- Quickly build and run the [sample CCF application](https://github.com/microsoft/ccf-app-samples)
```
```suggestion
- Get familiar with CCF core developer API with the [template CCF app](https://github.com/microsoft/ccf-app-template)
- Quickly build and run [sample CCF apps](https://github.com/microsoft/ccf-app-samples)
```

Just a suggestion to keep reference to both repos.

MahatiC reviewed (approved):
Thank you! LGTM

jumaffre commented on getting_started/azure_vm/install_ccf_on_azure_vm.sh:
```diff
@@ -0,0 +1,38 @@
+# Change these
+resourceGroup=rg_ccfdemo
+vm_name=ccfdemo
+vm_dns=ccfdemo-vm-sample
```
```suggestion
#!/bin/bash

# !!! Update these before running this script !!!
resourceGroup=rg_ccfdemo
vm_name=ccfdemo
vm_dns=ccfdemo-vm-sample
# !!!
```

jumaffre commented on doc/build_apps/install_bin.rst:
```diff
@@ -88,3 +88,13 @@ From Source
+In Azure
+--------
+
+You can install CCF on an Azure Virtual Machine by editing the ``install_ccf_on_azure_vm.sh`` and defining your own DNS name.
```
```suggestion
CCF can be installed on an Azure Virtual Machine by setting the resource group, VM and DNS names in the :ccf_repo:`getting_started/azure_vm/install_ccf_on_azure_vm.sh` script and running it.
```

FYI, the we use `ccf_repo` directive to link to the main repository so that links are versioned with the documentation.

jumaffre commented on getting_started/README.md:
```diff
@@ -0,0 +1,7 @@
+# Running CCF on a virtual machine
+
+This folder contains two separate ways of creating a Virtual Machine to run CCF. 
```
```suggestion
This folder contains two separate ways of setting up a development environment for CCF itself and apps.
```

jumaffre commented on getting_started/README.md:
```diff
@@ -0,0 +1,7 @@
+This folder contains two separate ways of creating a Virtual Machine to run CCF. 
+
+1. [azure_vm](./azure_vm/README.md) provides an opinionated way of creating a Virtual Machine on Azure and install CCF for you by cloud-init.
+1. setup_vm assumes you have cloned this repository onto your destination Virtual Machine and provides Ansible scripts to setup your VM
```
```suggestion
2. `setup_vm` assumes that a bare environment is already available (VM or container) and provides Ansible scripts to setup the development environment by either cloning this repository or installing the CCF Debian package
```

jumaffre commented on getting_started/azure_vm/install_ccf_on_azure_vm.sh:
```diff
@@ -0,0 +1,38 @@
+
+# These are defaults
+location=westeurope
+vm_size=Standard_DC1s_v2
```
This seems pretty small for the default, see https://learn.microsoft.com/en-us/azure/virtual-machines/dcv3-series#dcsv3-series. Perhaps `Standard_DC16s_v3` would be a better default here?

achamayou replied:
To run, rather than build CCF, this seems fine?

ross-p-smith replied:
I was able to run the banking sample on a DC1s. I've added a comment to tell people that this is the smallest VM. I didn't want users to pick a large machine and leave it running for a getting started

achamayou replied:
@ross-p-smith this is a good choice to get started, they can scale up if/when they need to.

jumaffre reviewed (approved):
@ross-p-smith Thank you for your contribution! I've suggested some minor edits to your PR, mostly for consistency with existing documentation (e.g. avoiding the use of "you"). Let us know if they make sense for your use case and we can get this PR merged.

achamayou commented:
@ross-p-smith can you run `./scripts/ci-checks.sh -f`? That will run the various formatters to make the format and license check pass.

achamayou commented:
/azp run

jumaffre commented:
@ross-p-smith The `CCF GitHub Pages` failure is due to the new link to the bash script that is broken because the file doesn't yet exist on the `main` branch. This is OK to merge regardless.

ross-p-smith commented:
> @ross-p-smith The `CCF GitHub Pages` failure is due to the new link to the bash script that is broken because the file doesn't yet exist on the `main` branch. This is OK to merge regardless.

Thanks @jumaffre! I've just removed the link as I didn't see your message. I've simplified the whole process by removing the dns so we dont need unique values in that script anymore. This means we don't really have to link people to the script anymore

achamayou commented:
/azp run
